src/constructor_helpers.cpp

```diff
@@ -198,6 +203,24 @@ vector<ArrayCoordinate> convert_to_polygon(Model<char>* model, ArrayCoordinate o
     return to_return;
 }
 
+vector<ArrayCoordinate> convert_to_polygon(Model<bool>* model, ArrayCoordinate offset, ArrayCoordinate edge_point){
+  Model<char>* char_model = new Model<char>(model->nrows(), model->ncols(), MODEL_UNSET);
```

Comment:
Modified code to use smaller vectors of points instead of the individual pit masks (Models that were the size of the entire grid_square for each pit). Should be much more efficient now.